simpleclient_common/src/test/java/io/prometheus/client/exporter/common/TextFormatTest.java

```diff
@@ -54,6 +57,30 @@ public void testCounterOutput() throws IOException {
                  + "nolabels 1.0\n", writer.toString());
   }
 
+  @Test
+  public void testMetricOutputWithTimestamp() throws IOException {
+    final Long now = System.currentTimeMillis();
+
+    class CustomCollector extends Collector {
+      public List<MetricFamilySamples> collect() {
+        List<MetricFamilySamples> mfs = new ArrayList<MetricFamilySamples>();
+        ArrayList<String> labelNames = new ArrayList<String>();
+        ArrayList<String> labelValues = new ArrayList<String>();
+        ArrayList<MetricFamilySamples.Sample> samples = new ArrayList<Collector.MetricFamilySamples.Sample>();
+        MetricFamilySamples.Sample sample = new MetricFamilySamples.Sample("nolabels", labelNames, labelValues, 1.0, now);
+        samples.add(sample);
+        mfs.add(new MetricFamilySamples("nolabels", Collector.Type.UNTYPED, "help", samples));
+        return mfs;
+      }
+    }
+    
+    new CustomCollector().register(registry);
+    TextFormat.write004(writer, registry.metricFamilySamples());
+    assertEquals("# HELP nolabels help\n"
+                 + "# TYPE nolabels untyped\n"
+                 + "nolabels 1.0 "+ now +"\n", writer.toString());
```

Comment:
I'd prefer this was a literal string rather than dynamic, as the output here depends on Java's implicit conversion to string.